kraftwerk-core/src/main/java/fr/insee/kraftwerk/core/parsers/LunaticXmlDataParser.java

```diff
@@ -204,18 +204,50 @@ private void readExternal(Element questionnaireNode, QuestionnaireData questionn
 
 	private void addGroupVariables(VariablesMap variables, String variableName, GroupInstance answers, Element node,
 			boolean collected) {
+
+		if (!variables.hasVariable(variableName)) {
+			log.debug("Variable not define : {}", variableName);
+			return;
+		}
+		manageTcmLiens(variableName, answers, node);
+		
+		String groupName = variables.getVariable(variableName).getGroupName();
+		GroupData groupData = answers.getSubGroup(groupName);
 		Elements valueNodes = node.getChildElements();
 
-		if (variables.hasVariable(variableName)) {
-			String groupName = variables.getVariable(variableName).getGroupName();
+		for (int j = 0; j < valueNodes.size(); j++) {
+			Element valueNode = valueNodes.get(j);
+			if (nodeExistsWithCompleteAttribute(valueNode)) {
+				String value = valueNodes.get(j).getValue();
+				if (collected) {
+					updateMaxLength(variables, variableName, value);
+				}
+				groupData.putValue(value, variableName, j);
+			}
+		}
+
+	}
+
+	private void manageTcmLiens(String variableName, GroupInstance answers, Element node) {
+		if (variableName.equals("LIENS")) {
+			String groupName = "BOUCLE_PRENOMS";
 			GroupData groupData = answers.getSubGroup(groupName);
-			for (int j = 0; j < valueNodes.size(); j++) {
-				Element valueNode = valueNodes.get(j);
-				if (nodeExistsWithCompleteAttribute(valueNode)) {
-					String value = valueNodes.get(j).getValue();
-					if (collected)
-						updateMaxLength(variables, variableName, value);
-					groupData.putValue(value, variableName, j);
+			Elements individualNodes = node.getChildElements();
+			
+			for (int j = 0; j < individualNodes.size(); j++) {
+				Element individualNode = individualNodes.get(j);
+				Elements valueNodes = individualNode.getChildElements();
+
+				for (int k=1;k <21;k++) {
+					String value = "99";
+					if (k<=valueNodes.size()) { 
+						value="0";
+						Element valueNode = valueNodes.get(k-1);
+						if (nodeExistsWithCompleteAttribute(valueNode)) {
+							value = valueNodes.get(k-1).getValue();
+						}
+					}
+					groupData.putValue(value, "LIEN_"+k, j);
```

Comment:
if  `"LIENS"`, `21`, `"LIEN_"`, `"BOUCLE_PRENOMS"` values are refactored, you could use the class that holds these here